core/server/master/src/main/java/alluxio/master/metastore/caching/CachingInodeStore.java

```diff
@@ -691,10 +697,13 @@ public void clear() {
       mEvictionHead = mMap.entrySet().iterator();
     }
 
-    private Map<String, Long> loadChildren(Long inodeId, ListingCacheEntry entry) {
-      evictIfNecessary();
+    private Map<String, Long> loadChildren(Long inodeId, ListingCacheEntry entry,
+        ReadOption option) {
+      if (!option.shouldSkipCache()) {
+        evictIfNecessary();
+      }
       entry.mModified = false;
-      Map<String, Long> listing = mEdgeCache.getChildIds(inodeId);
+      Map<String, Long> listing = mEdgeCache.getChildIds(inodeId, option);
       mMap.computeIfPresent(inodeId, (key, value) -> {
```

Comment:
Do we also want to skip caching in `getChildIds`?  inside the `mMap.computeIfPresent` is the place where we cache the listing results in the`mMap of ListingCache`.

Reply:
yes